interview-questions.md

```diff
@@ -1,43 +1,73 @@
-
-action: "GET" location: /students
 
-action: "POST" location: /students
 
-action: "GET" location: /students/new
-
-action: "GET" location: /students/2
+4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-action: "GET" location: /students/2/edit
+    action: "GET" location: /students        - displays a list of all students
+    action: "POST" location: /students       - creates a new instance of student
+    action: "GET" location: /students/new    - returns an HTML form for creating a new student
+    action: "GET" location: /students/2      - displays the student with :id 2
+    action: "GET" location: /students/2/edit - returns an HTML form for editing the student with :id 2
+    action: "PATCH" location: /students/2    - updates the student with :id 2
+    action: "DELETE" location: /students/2   - deletes the student with :id 2
 
-action: "PATCH" location: /students/2
 
-action: "DELETE" location: /students/2
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
+    1. As myself, I want to categorize tasks by importance/priority, so I may better organize my list and choose to do either one really big and important task or do multiple smaller and less important tasks.
+    2. As myself, I want to give a deadline to my tasks, so I can feel a sense of urgency and push myself to complete those tasks.
+    3. As myself, I want to have the option to add prerequisites to tasks, so I know if there are other things to be completed before something else is done.
+    4. As myself, I want to mark tasks as complete, so I can check off things I've done.
+    5. As myself, I want to categorize tasks by status, so I can say if something is either in progress, hasn't gotten started yet, dropped, or completed.
+    6. As myself, I want to have the option to add a location to tasks, so I know if something should be done elsewhere (ex. Oil change for car @Valvoline)
+    7. As myself, I want to have the option to drop/abandon a task but still see it in my database, so I can see previous tasks and get an general overview of productivity.
+    8. As myself, I want to sort tasks by any chosen metric, so I can see them laid out by either importance, deadline, status, location, or by needing prerequisites or not.
+    9. As myself, I want to invite other users of the application to the task list, so we can collaborate on things needing to be done (ex. Create a new task list for house chores, and include everyone inside of the house on that list).
+    10. As myself, I want to designate tasks to certain people if it is a list with more than one person on it, so we can know who will be doing what.
```

Comment:
These are great developer stories but not great User stories. A User story expresses what can be seen, and done by the user with _implications_ for what needs to be built to accomplish that goal.